EEJT-0003/src/main/java/com/epam/esm/exceptionhandler/handler/ResponseExceptionHandler.java

```diff
@@ -0,0 +1,49 @@
+package com.epam.esm.exceptionhandler.handler;
+
+
+import com.epam.esm.exceptionhandler.exception.ApplicationException;
+import com.epam.esm.exceptionhandler.exception.ItemNotFoundException;
+import com.epam.esm.exceptionhandler.exception.ServerException;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+import java.util.Map;
+
+@ControllerAdvice
+public class ResponseExceptionHandler extends ResponseEntityExceptionHandler {
+    private static final String RESPONSE_BODY = "response body";
+    private static final String HTTP_STATUS = "HTTP Status";
+    private static final String MESSAGE = "message";
+    private static final String MESSAGE_FOR_APPLICATION_EXCEPTION = "Oops, something went wrong";
+
+    private Map<String, ?> buildResponseMap(HttpStatus status, String message) {
+        return Map.of(HTTP_STATUS, "" + status, RESPONSE_BODY, Map.of(MESSAGE, message));
+    }
+
+    @ExceptionHandler(ItemNotFoundException.class)
+    protected ResponseEntity<?> handleItemNotFoundException(ItemNotFoundException ex, WebRequest request) {
+
+        return handleExceptionInternal(ex,
+                buildResponseMap(HttpStatus.NOT_FOUND, ex.getMessage()),
+                new HttpHeaders(), HttpStatus.NOT_FOUND, request);
+    }
```

Comment:
You inherit one exception class from another, but still process it on the controller, maybe you should pass the status as a paramater to the parent class constructor?

Reply:
i don`t understand sorry